### PR TITLE
Add concept exercise "vehicle-purchase" (#1934)

### PR DESCRIPTION
**Pull request by kahgoh**
Exercise covers `conditional-if` and `comparison` concepts and is based on the [Javascript](https://github.com/exercism/javascript/tree/main/exercises/concept/vehicle-purchase) version.

### CODE REVIEW

**Comment by kahgoh** [comment]
>  The Concept Exercise `vehicle-purchase` has `comparison` in its `concepts`, but that concept appears in the `concepts` of another Concept Exercise:
> /home/runner/work/go/go/config.json
> 
> The Concept Exercise `vehicle-purchase` has `conditionals-if` in its `concepts`, but that concept appears in the `concepts` of another Concept Exercise:
> /home/runner/work/go/go/config.json

With this one, should I just remove the concepts for now? At least until the other task to remove the `comparison` and `conditional-if` from `cars-assemble` is done.

File: exercises/concept/vehicle-purchase/vehicle_purchase_test.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+			name:          "price is reduced to 50%% for age 11",
+			originalPrice: 50000,
+			age:           11,
+			expected:      25000,
+		},
+	}
+
+	for _, test := range tests {
+		actual := CalculateResellPrice(test.originalPrice, test.age)
+		if actual != test.expected {
```
Comment (andrerfcsantos) [inline exercises/concept/vehicle-purchase/vehicle_purchase_test.go:140]: Wondering if this is safe, as it is a float comparison. We should run the tests against more exemplar solutions to see if this breaks in any way. If it does, we probably need something like we did in [interest is interesting](https://github.com/exercism/go/blob/main/exercises/concept/interest-is-interesting/interest_is_interesting_test.go).

**Review by andrerfcsantos — CHANGES_REQUESTED** [review]
Thanks for this!

Did some initial review.

**Comment by andrerfcsantos** [comment]
> > The Concept Exercise `vehicle-purchase` has `comparison` in its `concepts`, but that concept appears in the `concepts` of another Concept Exercise:
> > /home/runner/work/go/go/config.json
> > The Concept Exercise `vehicle-purchase` has `conditionals-if` in its `concepts`, but that concept appears in the `concepts` of another Concept Exercise:
> > /home/runner/work/go/go/config.json
> 
> With this one, should I just remove the concepts for now? At least until the other task to remove the `comparison` and `conditional-if` from `cars-assemble` is done.

I'd say to remove them from `cars-assemble` for now. What do you think @junedev ?

**Comment by junedev** [comment]
@kahgoh Regarding the CI failure: Remove the two concepts from `cars-assemble` so they only appear for this new exercise. It's a little weird that those not unlock for cars assemble then but it's acceptable until the other issue is addressed.

File: exercises/concept/vehicle-purchase/vehicle_purchase_test.go (hunk 1 above)
Comment (kahgoh) [reply exercises/concept/vehicle-purchase/vehicle_purchase_test.go:140]: Yeah, I was able to find an example where this happens. I'll implement the same thing from the "interest is interesting" exercise.

**Comment by kahgoh** [comment]
```
 tmp/lint/concept/vehicle_purchase/vehicle_purchase.go:22:2: ifElseChain: rewrite if-else to switch statement (gocritic)
	if age < 3 {
	^
deleting tmp/lint
Error: Process completed with exit code 1.
```

I'm thinking we don't want to modify the exemplar to use `switch`, which is covered by the [blackjack](https://exercism.org/tracks/go/exercises/blackjack) exercise. Is there a way to disable this check for this exercise?

**Comment by junedev** [comment]
@kahgoh Yes, you can disable just that one linter for that one code block using a nolint directive. You can find the details here: https://golangci-lint.run/usage/false-positives/#nolint

File: exercises/concept/vehicle-purchase/.meta/exemplar.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ func ChooseVehicle(first, second string) string {
 // CalculateResellPrice calculates how much a vehicle can resell for at a certain age.
 func CalculateResellPrice(originalPrice, age float64) float64 {
 	var percentage float64
+	//nolint:gocritic // To disable ifElseChain. The point of this exercise is to teach if-else-if-else, not switch.
```
Comment (kahgoh) [inline exercises/concept/vehicle-purchase/.meta/exemplar.go:22]: The only way to work around the `ifElseChain` warning was to disable the `gocritic` checks. I wasn't sure if there was a way to disable just `ifElseChain` specifically? 

I tried `//nolint:gocritic,ifElseChain`, but I kept getting the warning `WARN [runner/nolint] Found unknown linters in //nolint directives: ifelsechain`

Comment (junedev) [reply exercises/concept/vehicle-purchase/.meta/exemplar.go:22]: Not that I know of.

File: exercises/concept/vehicle-purchase/.docs/instructions.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# Instructions
+
+In this exercise you are going to write some code to help you prepare to buy a vehicle.
+
+You have three tasks, one to determine if you need a licence, one to help you choose between two vehicles and one to estimate the acceptable price for a used vehicle.
+
+## 1. Determine if you will need a driver's licence
+
+Some vehicle kinds require a driver's licence to operate them.
+Assume only the kinds `'car'` and `'truck'` require a licence, everything else can be operated without a licence.
```
Comment (andrerfcsantos) [inline exercises/concept/vehicle-purchase/.docs/instructions.md:10]:
```suggestion
Assume only the kinds `"car"` and `"truck"` require a licence, everything else can be operated without a licence.
```

File: exercises/concept/vehicle-purchase/.docs/instructions.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+Implement the `NeedsLicence(kind)` function that takes the kind of vehicle and returns a boolean indicating whether you need a licence for that kind of vehicle.
+
+```go
+needLicence := NeedsLicence('car')
```
Comment (andrerfcsantos) [inline exercises/concept/vehicle-purchase/.docs/instructions.md:15]:
```suggestion
needLicence := NeedsLicence("car")
```

File: exercises/concept/vehicle-purchase/.docs/instructions.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+// Output: true
+
+needLicence = NeedsLicence('bike')
```
Comment (andrerfcsantos) [inline exercises/concept/vehicle-purchase/.docs/instructions.md:18]:
```suggestion
needLicence = NeedsLicence("bike")
```

File: exercises/concept/vehicle-purchase/.docs/instructions.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+## 2. Choose between two potential vehicles to buy
+
+You evaluate your options of available vehicles.
+You manage to narrow it down to two options but you need help making the final decision.
+For that implement the function `ChooseVehicle(option1, option2)` that takes two vehicles as arguments and returns a decision that includes the option that comes first in dictionary order.
+
+```go
+vehicle := ChooseVehicle('Wuling Hongguang', 'Toyota Corolla')
```
Comment (andrerfcsantos) [inline exercises/concept/vehicle-purchase/.docs/instructions.md:29]:
```suggestion
vehicle := ChooseVehicle("Wuling Hongguang", "Toyota Corolla")
```

File: exercises/concept/vehicle-purchase/.docs/instructions.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+// Output: "Toyota Corolla is clearly the better choice."
+
+ChooseVehicle('Volkswagen Beetle', 'Volkswagen Golf')
```
Comment (andrerfcsantos) [inline exercises/concept/vehicle-purchase/.docs/instructions.md:32]:
```suggestion
ChooseVehicle("Volkswagen Beetle", "Volkswagen Golf")
```

File: exercises/concept/vehicle-purchase/.docs/instructions.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+// Output: 800
+
+CalculateResellPrice(1000, 5)
+// Output: 700
+
+CalculateResellPrice(1000, 15)
+// Output: 500
+```
+
+Note the value returned is a `float64`.
```
Comment (andrerfcsantos) [inline exercises/concept/vehicle-purchase/.docs/instructions.md:58]:
```suggestion
**Note:** the value returned is a `float64`.
```

File: exercises/concept/vehicle-purchase/.meta/exemplar.go (hunk 2 above)
Comment (andrerfcsantos) [reply exercises/concept/vehicle-purchase/.meta/exemplar.go:22]:
```suggestion
	// Disables ifElseChain rule of gocritic, which suggests convert the following code to a switch,
	// however the point of this exercise is to teach if-else-if-else, not switch.
	//nolint:gocritic
```

File: exercises/concept/vehicle-purchase/.docs/instructions.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+needLicence = NeedsLicence('bike')
+// Output: false
+```
+
+## 2. Choose between two potential vehicles to buy
+
+You evaluate your options of available vehicles.
+You manage to narrow it down to two options but you need help making the final decision.
+For that implement the function `ChooseVehicle(option1, option2)` that takes two vehicles as arguments and returns a decision that includes the option that comes first in dictionary order.
```
Comment (andrerfcsantos) [inline exercises/concept/vehicle-purchase/.docs/instructions.md:26]:
```suggestion
For that, implement the function `ChooseVehicle(option1, option2)` that takes two vehicles as arguments and returns a decision that includes the option that comes first in dictionary order.
```

File: exercises/concept/vehicle-purchase/.docs/instructions.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+vehicle := ChooseVehicle('Wuling Hongguang', 'Toyota Corolla')
+// Output: "Toyota Corolla is clearly the better choice."
+
+ChooseVehicle('Volkswagen Beetle', 'Volkswagen Golf')
+// Output: "Volkswagen Beetle is clearly the better choice."
+```
+
+## 3. Calculate an estimation for the price of a used vehicle
+
+Now that you made your decision you want to make sure you get a fair price at the dealership.
```
Comment (andrerfcsantos) [inline exercises/concept/vehicle-purchase/.docs/instructions.md:38]:
```suggestion
Now that you made a decision, you want to make sure you get a fair price at the dealership.
```

**Review by andrerfcsantos — APPROVED** [review]
Took a closer look at all the files and left some additional suggestions, everything else looks good.

The main thing was to convert strings quoted with `'` to `"`, as strings with `'` are not valid Go strings.

**Review by andrerfcsantos — APPROVED** [review]

**Comment by andrerfcsantos** [comment]
The exercise is now live 🚀

**Comment by jkisor** [comment]
Hi, cool exercise! I did happen to run into an issue in which a test or the instructions need to be corrected.

```
{
  name:          "price is reduced to 70%% for age 10",
  originalPrice: 25000,
  age:           10,
  expected:      12500,
},
```

12500 / 25000 is 0.5 not 0.7 as the test name suggests.

**Comment by andrerfcsantos** [comment]
@jkisor Can you open an issue for that please?

**Comment by kahgoh** [comment]
Oops, I have raised PR #1964 without opening new issue

**Pull request merged**